### PR TITLE
feat(typing): add possibly-undefined-check to issues team / error ingestion critical code

### PR DESCRIPTION
**Pull request by JoshFerge**
We continually face incidents/issues where this type check configuration would have caught the error.

I attempted to add this for all of sentry in https://github.com/getsentry/sentry/pull/65589 but this was deemed unsatisfactory because there was a push to try and fix all of the errors in sentry before merging, a large task which i did not have time for.

Here, instead, we add an _allowlist_, which right now contains critical issues team code, which we want to have a higher safety bar for.

This PR _will_ prevent incidents, and make our ingestion safer. 

I fix one instance of this error in post_process.

### CODE REVIEW

**Review by vartec — APPROVED** [review]

File: pyproject.toml
Comment (asottile-sentry) [inline pyproject.toml:659]: if we're going to do this (I still don't think this is the right way to land this change and doesn't prevent this issue from happening in general) let's just add this to the `stronger` list above

**Review by asottile-sentry — CHANGES_REQUESTED** [review]

File: src/sentry/tasks/post_process.py
Comment (asottile-sentry) [inline src/sentry/tasks/post_process.py:1074]: slightly more correct / less error prone would be to use `try:` `else:`

File: pyproject.toml
Comment (JoshFerge) [reply pyproject.toml:659]: https://github.com/getsentry/sentry/pull/65589#discussion_r1499348693 -- on the previous PR, you asked me to make changes, i made the changes, and you still did not approve it, and when i made follow up comments, you did not address them or respond to them.

Here, can you please inform me that if i make these changes, you will approve the PR?

Comment (armenzg) [reply pyproject.toml:659]: Yes, let's do that.

It may also be easier to reduce the number of modules you want to adjust.

Comment (armenzg) [reply pyproject.toml:659]: Was possibly undefined a case missed which caused an incident?

**Review by armenzg — APPROVED** [review]
After chatting with Josh I believe this makes sense.

**Pull request closed without merging**
